### ISSUE TITLE
[Bug]: Extra growing segments appear in queryNode

### ISSUE THREAD

**Issue by ThreadDao** [milvus-io/milvus#12131]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master-20211118-27715c1
- Deployment mode(standalone or cluster): cluster
- SDK version(e.g. pymilvus v2.0.0rc2): pymilvus 2.0.0rc9.dev11
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others: dataCoord.enableCompaction: true
          dataCoord.compaction.retentionDuration: 20
```

### Current Behavior

1. Create a collection with `shard_num=1`
2. Insert one entity and flush (repeat five times, get 5 sealed segments)
3. Load collection
4. Get query segment info: 5 sealed segments and 2 growing segments
```
get_query_segment_info('compact_XLmAnuVd')
[segmentID: 429206898403966977
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 524
num_rows: 1
nodeID: 7
state: Sealed
, segmentID: 429206896464101377
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 524
num_rows: 1
nodeID: 7
state: Sealed
, segmentID: 429206899229720577
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 524
num_rows: 1
nodeID: 7
state: Sealed
, segmentID: 429206895638347777
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 524
num_rows: 1
nodeID: 7
state: Sealed
, segmentID: 429206897567465473
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 524
num_rows: 1
nodeID: 7
state: Sealed
, segmentID: 429206898403966977
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 17727488
num_rows: 1
nodeID: 7
state: Growing
, segmentID: 429206899229720577
collectionID: 429206895585918977
partitionID: 429206895585918978
mem_size: 17727488
num_rows: 1
nodeID: 7
state: Growing
]
```
**Logs:**  
[milvus_logs.tar.gz](https://github.com/milvus-io/milvus/files/7568037/milvus_logs.tar.gz)

### Expected Behavior

Only 5 sealed segments.

### Steps To Reproduce

```markdown
less_threshold_segments_num = 5
        collection_w = self.init_collection_wrap(name=cf.gen_unique_str(prefix), shards_num=1)
        for i in range(less_threshold_segments_num):
            df = cf.gen_default_dataframe_data(1)
            collection_w.insert(df)
            log.debug(collection_w.num_entities)
        sleep(3)
        collection_w.load()
        segments_info = self.utility_wrap.get_query_segment_info(collection_w.name)[0]
        assert len(segments_info) == less_threshold_segments_num
```
```


### Anything else?

_No response_

**Comment by congqixia** [comment]
The root cause is that the segments generated in test cases is compacted by Compaction. Yet the query cluster does not read the information when perform `Load` operations. So the compacted segments behind Checkpoints will become a new Growing Segment. @sunby @xige-16

**Comment by yanliang567** [comment]
2 quetstions:
1. why the num_rows=1 for the 2 growing segments? 
2. why it generated 2 growing segments, while we configured only 1 shard

**Comment by ThreadDao** [comment]
@congqixia In this case, `compact` is not called manually. At the same time, the number of segments is 5, which is less than the threshold (10) for automatically triggering the merge. I don't understand the root cause you said.

**Comment by binbinlv** [comment]
For statistics with other blocking issues for GA, so set this as critical urgent too.

**Comment by xige-16** [comment]
First question: querynode should add execlude segment for flow graph using flushed segment infos
```
	// add check points info
	checkPointInfos := make([]*datapb.SegmentInfo, 0)
	for _, info := range w.req.Infos {
		checkPointInfos = append(checkPointInfos, info.UnflushedSegments...)
	}
	w.node.streaming.replica.addExcludedSegments(collectionID, checkPointInfos)
```

**Comment by xige-16** [comment]
second question: all segments has been flushed, But there are still unfleshed segment infos in the returned recovery info
[2021/11/19 03:21:12.042 +00:00] [DEBUG] [collection_replica.go:181] ["excludedSegments in collectionReplica"] [info="{\"429206213382573377\":[{\"ID\":429206221286014979,\"collectionID\":429206213382573377,\"partitionID\":429206213382573378,\"dml_position\":{\"timestamp\":18446744073709551615}},{\"ID\":429206221299122177,\"collectionID\":429206213382573377,\"partitionID\":429206213382573378,\"dml_position\":{\"timestamp\":18446744073709551615}}]}"]

**Comment by congqixia** [comment]
> second question: all segments has been flushed, But there are still unfleshed segment infos in the returned recovery info [2021/11/19 03:21:12.042 +00:00] [DEBUG] [collection_replica.go:181] ["excludedSegments in collectionReplica"] [info="{"429206213382573377":[{"ID":429206221286014979,"collectionID":429206213382573377,"partitionID":429206213382573378,"dml_position":{"timestamp":18446744073709551615}},{"ID":429206221299122177,"collectionID":429206213382573377,"partitionID":429206213382573378,"dml_position":{"timestamp":18446744073709551615}}]}"]

@xige-16 Quick reply, this collection id is `429206213382573377`; Issue mentioned collection is `429206895585918977`;

**Comment by czs007** [comment]
@bigsheeper  is working on this.

**Comment by bigsheeper** [comment]
#12188 would fix the first question mentioned by @xige-16 above.

**Comment by bigsheeper** [comment]
#12247 have been merged， could you please help to retest it？@ThreadDao 
/unassign
/assign @ThreadDao

**Comment by ThreadDao** [comment]
Fixed `master-20211125-ad83090`
